src/vmod_dynamic.c

```diff
@@ -998,8 +1000,12 @@ vmod_director_backend(VRT_CTX, struct vmod_dynamic_director *obj,
 	if ((host == NULL || *host == '\0') && ctx->http_req != NULL)
 		host = VRT_GetHdr(ctx, &HDR_REQ_HOST);
 
-	if (host == NULL || *host == '\0')
+	if (host == NULL || *host == '\0') {
+		if (ctx->method == VCL_MET_INIT)
+			VRT_fail(ctx, "%s.backend(): host may not be empty "
```

Comment:
not sure.
`NULL` is now a valid backend (`backend None` in VCL), so why should we fail the vcl loading if anyone asked for a none-backend via vmod_dynamic?

Reply:
Yeah someone could use None as a sentinel for "no backend here", which is what it's for. And the empty string is a legal domain name. Can't imagine that anyone will ever configure the root domain as a backend, but we don't have to make it illegal if it doesn't break the VMOD.

Ok I'll try it without this restriction.